src/keboola/client/http.py

```diff
@@ -0,0 +1,448 @@
+
+        Args:
+            *args: Positional arguments to pass to the post request.
+                Accepts supported params in requests.sessions.Session#request
+            **kwargs: Key word arguments to pass to the post request.
+                Accepts supported params in requests.sessions.Session#request
+                eg. params = {'locId':'1'}, header = {some additional header}
+                parameters and headers are appended to the default ones
+            ignore_auth: True to skip authentication
+
+
+        Returns:
+            body: json reposonse json-encoded content of a response
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        kwargs['ignore_auth'] = ignore_auth
+        if params:
+            kwargs['params'] = params
+        kwargs['url'] = url
+        r = self.get_raw(**kwargs)
+        return r.json()
+
+    def post_raw(self, *args, **kwargs):
+        """
+        Construct a requests POST call with args and kwargs and process the
+        results.
+
+        Args:
+            *args: Positional arguments to pass to the post request.
+               Accepts supported params in requests.sessions.Session#request
+            **kwargs: Key word arguments to pass to the post request.
+               Accepts supported params in requests.sessions.Session#request
+               eg. params = {'locId':'1'}, header = {some additional header}
+               parameters and headers are appended to the default ones
+               ignore_auth: True to skip authentication
+
+        Returns:
+            Response: Returns :class:`Response <Response>` object.
+
+        """
+        method = 'POST'
+
+        r = self._request_raw(method, *args, **kwargs)
+        return r
+
+    def post(self, *args, ignore_auth=False, **kwargs):
+        """
+        Construct a requests POST call with args and kwargs and process the
+        results.
+
+        Args:
+            *args: Positional arguments to pass to the post request.
+               Accepts supported params in requests.sessions.Session#request
+            **kwargs: Key word arguments to pass to the post request.
+               Accepts supported params in requests.sessions.Session#request
+               eg. params = {'locId':'1'}, header = {some additional header}
+               parameters and headers are appended to the default ones
+            ignore_auth: True to skip authentication
+
+
+        Returns:
+            body: json reposonse json-encoded content of a response
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+
+        try:
+            kwargs['ignore_auth'] = ignore_auth
+            r = self.post_raw(*args, **kwargs)
+            r.raise_for_status()
+        except requests.HTTPError as e:
+            logging.warning(e, exc_info=True)
+            # Handle different error codes
+            raise
+        else:
+            return r.json()
```

Comment:
Apart from changing and unifying the signature, I think it may be a good idea to use decorators to do this `method` vs `method_raw `behaviour. Written like this it leads to code duplication and if we want to change the exception handling in the future, it will need to be changed in many places.

This is a usecase for decorators ([here's a nice article](https://realpython.com/primer-on-python-decorators/))

So we would have 

```python
post_raw():
  method = 'POST'
  
          r = self._request_raw(method, *args, **kwargs)
          return r
```

and post:

```python
@json_response_error_handling
def post(...):
   return post_raw()
```

and something like:
```python
def json_response_error_handling(func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        try:
            kwargs['ignore_auth'] = ignore_auth
            r = func(*args, **kwargs)
            r.raise_for_status()
        except requests.HTTPError as e:
            logging.warning(e, exc_info=True)
            # Handle different error codes
            raise
        else:
            return r.json()
    return wrapper
```

This way if we decide to refactor the error handling it will have to be done in one spot only.